### PR TITLE
feat(dropdown button): dropdown button adhering to Freshworks Design …

### LINKED ISSUE #112: [Feature Request] - Dropdown button
**Describe the functionality of the new component:**
A dropdown  button according to the Design system found in 
https://xd.adobe.com/view/a8299f86-4d90-40f9-4d23-7dc26f9266e3-de18/screen/eb6930f3-3b16-49ee-9aaa-3dbc780e7154/

**More details:**
Screenshot 2020-08-17 at 10 55 50 PM

### Properties
This will be part of fw-button itself
Property | Attribute | Description | Type | Default
---------|----------|------------|-------|--------
dropdown|dropdown|dropdown button or not|Boolean|false
split|split|Whether to split dropdown or not. Invalid if dropdown is false|Boolean|false
searchable|searchable|Whether the dropdown button should be searchable. Invalid if dropdown is false|Boolean|false
value|value|Selected options|Array|undefined
placeholder|placeholder|Placeholder for search input|String|''

### Events
Event | Description | Type
------|-------------|------
fwOptionClick|Emitted on selecting an option|customEvent
fwOptionsAdd|Emitted on adding options (`Add` button click)|customEvent

### Sample HTML
```html 
<fw-button dropdown split> Select OTT 
  <span id="1" value="netflix"> Netflix </span>
  <span id="2" value="amazon"> Amazon </span>
</fw-button>
```

### PR DESCRIPTION
**Pull request by shravan-balasubramanian**
…System

Dropdown button similar to Select component but build on top of fw-button. It can be searchable,
splittable and supports multi select

fix #112

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] My commits have standard messages as mentioned in [Contributing Guidelines](./../blob/next/CONTRIBUTING.md)
 
## How Has This Been Tested?
Storybook, e2e tests and stencil local testing

### CODE REVIEW

**Comment by sathishkumar-thangavel** [comment]
@shravan-balasubramanian build failing, unable run in dev. Please check it

File: src/components/dropdown-button/dropdown-button.tsx [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+      this.handleDropdownToggle();
+    }
+  }
+
+  render() {
+    const ChevronDown = () => {
+      const iconColor = ['secondary', 'link', 'text'].includes(this.color) ? '#12344d' : '#fff';
+      const direction = this.isDropdownOpen ? 'up' : 'down';
+
+      return <fw-icon name={`chevron-${direction}`} color={iconColor} size="8" > </fw-icon>;
```
Comment (asifahmedfw) [inline src/components/dropdown-button/dropdown-button.tsx:165]: Revert `fw-icon` size to a number. You can add it as a number here.

File: src/index.html [hunk 2]
```diff
@@ -48,6 +59,12 @@
...
   }
+
+  document.querySelector('fw-dropdown-button')
+    .addEventListener('fwOptionsAdd', (e) => console.log('Selected values are ', e.detail.value))
+
+  document.querySelector('fw-dropdown-button')
+    .addEventListener('fwOptionClick', (e) => console.log('Selected values are ', e.detail.value))
 </script>
-</body>
+</bod>
```
Comment (asifahmedfw) [inline src/index.html:69]: Typo 😄

File: src/components/dropdown-button/dropdown-button.tsx [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+    this.handleDropdownToggle();
+  }
+
+  /**
+   * Handles option select from the dropdown
+   */
+  private handleOptionClick(option) {
+    if (!this.searchable) {
+      this.value = option.value;
+      this.fwOptionClick.emit({ value: this.value });
```
Comment (asifahmedfw) [inline src/components/dropdown-button/dropdown-button.tsx:155]: Can you also send the text?

**Review by asifahmedfw — CHANGES_REQUESTED** [review]

File: src/components/dropdown-button/dropdown-button.tsx (hunk 1 above)
Comment (shravan-balasubramanian) [reply src/components/dropdown-button/dropdown-button.tsx:165]: I see an error for number. Anyways will check it and revert!

File: src/components/dropdown-button/dropdown-button.tsx (hunk 3 above)
Comment (shravan-balasubramanian) [reply src/components/dropdown-button/dropdown-button.tsx:155]: Label?. Done!

File: src/components/dropdown-button/dropdown-button.scss [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+  @prop --dropdown-min-height: Dropdown minimum height in pixels
+  @prop --dropdown-min-width: Dropdown minimum width in pixels
+  @prop --dropdown-max-height: Dropdown maximum height in pixels
+  @prop --dropdown-max-width: Dropdown maximum width in pixels
+  @prop --dropdown-font-size: Dropdown item font size in pixels
+*/
+
+//=== Button Style ===//
+:host {
+  --btn-primary-color: var(--color-milk);
```
Comment (asifahmedfw) [inline src/components/dropdown-button/dropdown-button.scss:13]: Are you using this?

**Review by asifahmedfw — APPROVED** [review]

**Pull request merged**